src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReport.java

```diff
@@ -0,0 +1,116 @@
+
+import com.microsoft.azure.servicebus.primitives.StringUtil;
+import org.apache.commons.collections4.CollectionUtils;
+import uk.gov.hmcts.ecm.common.helpers.UtilHelper;
+import uk.gov.hmcts.ecm.common.model.ccd.items.RespondentSumTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.types.ClaimantType;
+import uk.gov.hmcts.ecm.common.model.ccd.types.ClaimantWorkAddressType;
+import uk.gov.hmcts.ecm.common.model.reports.claimsbyhearingvenue.ClaimsByHearingVenueSubmitEvent;
+import uk.gov.hmcts.ethos.replacement.docmosis.helpers.ReportHelper;
+import uk.gov.hmcts.ethos.replacement.docmosis.reports.ReportParams;
+import java.util.Comparator;
+import java.util.List;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.CLAIMS_BY_HEARING_VENUE_REPORT;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.SINGLE_HEARING_DATE_TYPE;
+
+public class ClaimsByHearingVenueReport {
+    private final ClaimsByHearingVenueReportDataSource reportDataSource;
+    private final ReportParams reportParams;
+    private static final String NULL_STRING_VALUE = "Null";
+
+    public ClaimsByHearingVenueReport(ClaimsByHearingVenueReportDataSource dataSource, ReportParams params) {
+        reportDataSource = dataSource;
+        reportParams = params;
+    }
+
+    public ClaimsByHearingVenueReportData generateReport(String listingDate, String hearingDateType) {
+        var submitEvents = reportDataSource.getData(
+            UtilHelper.getListingCaseTypeId(reportParams.getCaseTypeId()),
+            reportParams.getDateFrom(), reportParams.getDateTo());
+        var claimsByHearingVenueReportData = initReport(reportParams.getCaseTypeId());
+        setReportListingDate(claimsByHearingVenueReportData, listingDate,
+            reportParams.getDateFrom(), reportParams.getDateTo(), hearingDateType);
+
+        if (CollectionUtils.isNotEmpty(submitEvents)) {
+            setReportData(submitEvents, claimsByHearingVenueReportData);
+        }
+        return claimsByHearingVenueReportData;
+    }
+
+    private ClaimsByHearingVenueReportData initReport(String caseTypeId) {
+        var reportData =  new ClaimsByHearingVenueReportData();
+        reportData.setOffice(UtilHelper.getListingCaseTypeId(caseTypeId));
+        reportData.setReportType(CLAIMS_BY_HEARING_VENUE_REPORT);
+        reportData.setDocumentName(CLAIMS_BY_HEARING_VENUE_REPORT);
+        return reportData;
+    }
+
+    private void setReportListingDate(ClaimsByHearingVenueReportData reportData, String listingDate,
+                                      String listingDateFrom, String listingDateTo, String hearingDateType) {
+        if (SINGLE_HEARING_DATE_TYPE.equals(hearingDateType)) {
+            reportData.setListingDate(listingDate);
+            reportData.setListingDateFrom(null);
+            reportData.setListingDateTo(null);
+            reportData.setHearingDateType(hearingDateType);
+            reportData.setReportPeriodDescription("On "
```

Comment:
Period description can be defined in template directly I think. Please check other report templates. It would be nice to use them for consistency

Reply:
This report does not use template as its output is Excel.